qt/scientific_interfaces/Indirect/IndirectSymmetriseView.cpp

```diff
@@ -0,0 +1,434 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "IndirectSymmetriseView.h"
+#include "IndirectDataValidationHelper.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidQtWidgets/Common/UserInputValidator.h"
+#include "MantidQtWidgets/Plotting/SingleSelector.h"
+
+#include <QDoubleValidator>
+#include <QFileInfo>
+
+using namespace IndirectDataValidationHelper;
+using namespace Mantid::API;
+
+namespace {
+
+QPair<double, double> getXRangeFromWorkspace(const Mantid::API::MatrixWorkspace_const_sptr &workspace) {
+  auto const xValues = workspace->x(0);
+  return QPair<double, double>(xValues.front(), xValues.back());
+}
+} // namespace
+
+using MantidQt::MantidWidgets::AxisID;
+namespace MantidQt::CustomInterfaces {
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+IndirectSymmetriseView::IndirectSymmetriseView(QWidget *parent) {
+  m_uiForm.setupUi(parent);
+  m_dblManager = new QtDoublePropertyManager();
+  m_dblEdFac = new DoubleEditorFactory(this);
```

Comment:
I only knew what `EdFac` was because it was next to the definition. Could we please have a better name?